src/common/freertos_mutex.hpp

```diff
@@ -29,3 +29,19 @@ class FreeRTOS_Mutex {
     void *xSemaphore = nullptr;     // SemaphoreHandle_t
     char xMutexBuffer[buffer_size]; // this size is checked in cpp file
 };
+
+// we need our own lock, because GCC uses try and back off which internally uses try_lock and because FreeRTOS does not use priority inheritance when trying to lock mutex without blocking
+// this results in deadlock (inverse priority problem), because thread with lower priority does not get CPU time while still holding mutex.
+namespace buddy {
+
+template <typename lockable1, typename lockable2>
+void lock(lockable1 &l1, lockable2 &l2) {
+    if (&l1 < &l2) {
```

Comment:
This probably might break if caller once wrap pointer to mutex in some class with `lock` method. And in one place it locks wrapper class and in different place bare mutex. Maybe it could be solved if this is not template but type of object it operates on is enforced.